lib/ansible/modules/cloud/azure/azure_rm_webapp.py

```diff
@@ -0,0 +1,971 @@
+          name: myappplan
+        app_settings:
+          testkey: testvalue
+        frameworks:
+          - name: "java"
+            version: "1.8"
+            settings:
+            - name: "java_container"
+              value: "Tomcat"
+            - name: "java_container_version"
+              value: "8.0"
+'''
+
+RETURN = '''
+azure_webapp:
+    description: Facts about the current state of the web app.
+    returned: always
+    type: dict
+    sample: {
+        "availability_state": "Normal",
+        "client_affinity_enabled": true,
+        "client_cert_enabled": false,
+        "container_size": 0,
+        "daily_memory_time_quota": 0,
+        "default_host_name": "ansiblewindowsaaa.azurewebsites.net",
+        "enabled": true,
+        "enabled_host_names": [
+            "ansiblewindowsaaa.azurewebsites.net",
+            "ansiblewindowsaaa.scm.azurewebsites.net"
+        ],
+        "host_name_ssl_states": [
+            {
+                "host_type": "Standard",
+                "name": "ansiblewindowsaaa.azurewebsites.net",
+                "ssl_state": "Disabled"
+            },
+            {
+                "host_type": "Repository",
+                "name": "ansiblewindowsaaa.scm.azurewebsites.net",
+                "ssl_state": "Disabled"
+            }
+        ],
+        "host_names": [
+            "ansiblewindowsaaa.azurewebsites.net"
+        ],
+        "host_names_disabled": false,
+        "id": "/subscriptions/<subscription_id>/resourceGroups/ansiblewebapp1/providers/Microsoft.Web/sites/ansiblewindowsaaa",
+        "kind": "app",
+        "last_modified_time_utc": "2018-05-14T04:50:54.473333Z",
+        "location": "East US",
+        "name": "ansiblewindowsaaa",
+        "outbound_ip_addresses": "52.170.7.25,52.168.75.147,52.179.5.98,52.179.1.81,52.179.4.232",
+        "repository_site_name": "ansiblewindowsaaa",
+        "reserved": false,
+        "resource_group": "ansiblewebapp1",
+        "scm_site_also_stopped": false,
+        "server_farm_id": "/subscriptions/<subscription_id>/resourceGroups/test/providers/Microsoft.Web/serverfarms/plan1",
+        "state": "Running",
+        "tags": {},
+        "type": "Microsoft.Web/sites",
+        "usage_state": "Normal"
+    }
+'''
+
+import time
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from msrest.serialization import Model
+    from azure.mgmt.web.models import (
+        site_config, app_service_plan, Site,
+        AppServicePlan, SkuDescription, NameValuePair
+    )
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+container_settings_spec = dict(
+    name=dict(type='str', required=True),
+    registry_server_url=dict(type='str'),
+    registry_server_user=dict(type='str'),
+    registry_server_password=dict(type='str')
```

Comment:
Set `no_log=True` for any password/secret fields to stop Ansible from logging them or displaying them to the user.

Reply:
resolved